### PR TITLE
fix: set access-control-allow-credentials=true for auth.pl

### PR DESCRIPTION
**Pull request by stephanegigandet**
Fix for https://github.com/openfoodfacts/openfoodfacts-server/issues/6462

### CODE REVIEW

File: cgi/auth.pl [hunk 1]
```diff
@@ -47,7 +47,14 @@
...
 }
 
-print header ( -status => $status );
+print header(-status => $status);
+
+# We need to send the header Access-Control-Allow-Credentials=true so that websites
+# such has hunger.openfoodfacts.org that send a query to world.openfoodfacts.org/cgi/auth.pl
+# can read the resulting response.
+
+print header(-Access_Control_Allow_Credentials => "true");
```
Comment (alexgarel) [inline cgi/auth.pl:56]: As we are sending the header from here, couldn't we first check for referer and limit this to:

`defined env["HTTP_REFERER"] and env["HTTP_REFERER"] =~ /^(?:https?://)\w+$main_domain/`

assuming $main_domain contains openfoodfacts.org/net/localhost

**Review by alexgarel — APPROVED** [review]
LGTM - proposed an improvement, but not mandatory

File: cgi/auth.pl (hunk 1 above)
Comment (stephanegigandet) [reply cgi/auth.pl]: Good idea, I added a change like this. The browser will send an Origin header.

**Comment by alexgarel** [comment]
ok, I imagine you will test before merging ?

**Comment by alexgarel** [comment]
@stephanegigandet is this ready to merge ?

**Pull request merged**